File: go/2697.go
package main

func makeSmallestPalindrome(s string) string {
	str := []byte(s)
	i := 0
	j := len(str) - 1
	for i < j {
		if str[i] != str[j] {
			if str[i] < str[j] {
				str[j] = str[i]
			} else {
				str[i] = str[j]
			}
		}
		i++
		j--
	}
	return string(str)
}

File: python/2697.py
class Solution:
    def makeSmallestPalindrome(self, s: str) -> str:
        strs = list(s)
        l, r = 0, len(strs)-1
        while l < r:
            if strs[l] != strs[r]:
                if ord(strs[l]) < ord(strs[r]):
                    strs[r] = strs[l]
                else:
                    strs[l] = strs[r]
            l += 1
            r -= 1
        return "".join(strs)